core/tests/test_core_ports.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+from typing import Union, Optional
+from testcontainers.core.container import DockerContainer
+
+from docker.errors import APIError
+
+
+@pytest.mark.parametrize(
+    "container_port, host_port",
+    [
+        ("8080", "8080"),
+        ("8125/udp", "8125/udp"),
+        ("8092/udp", "8092/udp"),
+        ("9000/tcp", "9000/tcp"),
+        ("8080", "8080/udp"),
+        (8080, 8080),
+        (9000, None),
+        ("9009", None),
+        ("9000", ""),
+        ("9000/udp", ""),
+    ],
+)
+def test_docker_container_with_bind_ports(container_port: Union[str, int], host_port: Optional[Union[str, int]]):
+    container = DockerContainer("alpine:latest")
+    container.with_bind_ports(container_port, host_port)
+    container.start()
+
+    container_id = container._container.id
+    client = container._container.client
+
+    if isinstance(container_port, int):
+        container_port = str(container_port)
+    if isinstance(host_port, int):
+        host_port = str(host_port)
+    if not host_port:
+        host_port = ""
+
+    # if the port protocol is not specified, it will default to tcp
+    if "/" not in container_port:
+        container_port += "/tcp"
+
+    excepted = {container_port: [{"HostIp": "", "HostPort": host_port}]}
+    assert client.containers.get(container_id).attrs["HostConfig"]["PortBindings"] == excepted
+    container.stop()
```

Comment:
readability suggestion

```python
def test_docker_container_with_bind_ports(container_port: Union[str, int], host_port: Optional[Union[str, int]]):
    container = DockerContainer("alpine:latest")
    container.with_bind_ports(container_port, host_port)
    container.start()

    # prepare to inspect container
    container_id = container._container.id
    client = container._container.client

    # assemble expected output to compare to container API
    container_port = str(container_port)
    host_port = str(host_port or "")

    # if the port protocol is not specified, it will default to tcp
    if "/" not in container_port:
        container_port += "/tcp"

    expected = {container_port: [{"HostIp": "", "HostPort": host_port}]}  # typo (expected)

    # compare PortBindings to expected output
    assert client.containers.get(container_id).attrs["HostConfig"]["PortBindings"] == expected
    container.stop()
```

?

Reply:
super nice, thanks will update